tests/lib/EventDispatcher/EventListener/ContentSuggestionSubscriberTest.php

```diff
@@ -8,67 +8,161 @@
 
+use Ibexa\Bundle\Core\ApiLoader\RepositoryConfigurationProvider;
+use Ibexa\Contracts\Core\Repository\SearchService as SearchServiceInterface;
 use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Ibexa\Contracts\Core\Repository\Values\Content\Query;
 use Ibexa\Contracts\Core\Repository\Values\Content\Search\SearchHit;
 use Ibexa\Contracts\Core\Repository\Values\Content\Search\SearchResult;
 use Ibexa\Contracts\Core\Repository\Values\ContentType\ContentType;
 use Ibexa\Contracts\Search\Event\BuildSuggestionCollectionEvent;
 use Ibexa\Contracts\Search\Mapper\SearchHitToContentSuggestionMapperInterface;
 use Ibexa\Contracts\Search\Model\Suggestion\ContentSuggestion as ContentSuggestionModel;
+use Ibexa\Contracts\Search\Model\Suggestion\Suggestion;
+use Ibexa\Core\Base\Exceptions\InvalidArgumentException;
 use Ibexa\Core\Repository\SiteAccessAware\SearchService;
 use Ibexa\Core\Repository\Values\Content\Location;
 use Ibexa\Search\EventDispatcher\EventListener\ContentSuggestionSubscriber;
 use Ibexa\Search\Model\SuggestionQuery;
 use PHPUnit\Framework\TestCase;
+use Psr\Log\LoggerInterface;
 
 final class ContentSuggestionSubscriberTest extends TestCase
 {
+    /** @var \Ibexa\Bundle\Core\ApiLoader\RepositoryConfigurationProvider|(RepositoryConfigurationProvider&object&\PHPUnit\Framework\MockObject\MockObject)|(RepositoryConfigurationProvider&\PHPUnit\Framework\MockObject\MockObject)|(object&\PHPUnit\Framework\MockObject\MockObject)|\PHPUnit\Framework\MockObject\MockObject */
+    private RepositoryConfigurationProvider $configProviderMock;
+
+    private ?Query $capturedQuery;
+
+    private bool $searchServiceSupportsScoring = false;
+
+    /**
+     * @var \Psr\Log\LoggerInterface|(object&\PHPUnit\Framework\MockObject\MockObject)|\PHPUnit\Framework\MockObject\MockObject|(LoggerInterface&object&\PHPUnit\Framework\MockObject\MockObject)|(LoggerInterface&\PHPUnit\Framework\MockObject\MockObject)
+     */
+    private LoggerInterface $loggerMock;
+
+    protected function setUp(): void
+    {
+        $this->configProviderMock = $this->createMock(RepositoryConfigurationProvider::class);
+        $this->capturedQuery = null;
+        $this->searchServiceSupportsScoring = false;
+        $this->loggerMock = $this->createMock(LoggerInterface::class);
+    }
+
     public function testSubscribedEvents(): void
     {
-        $this->assertSame(
-            [BuildSuggestionCollectionEvent::class => 'onBuildSuggestionCollectionEvent'],
+        self::assertSame(
+            [
+                BuildSuggestionCollectionEvent::class => 'onBuildSuggestionCollectionEvent',
+            ],
             ContentSuggestionSubscriber::getSubscribedEvents()
         );
     }
 
-    public function testOnContentSuggestion(): void
+    public function testOnContentSuggestionWithLegacyEngineAndScoring(): void
+    {
+        $this->configProviderMock->method('getRepositoryConfig')
+            ->willReturn(['search' => ['engine' => 'legacy']]);
```

Comment:
```suggestion
        $this->configProviderMock
            ->method('getRepositoryConfig')
            ->willReturn(['search' => ['engine' => 'legacy']]);
```
or
```suggestion
        $this->configProviderMock->method('getRepositoryConfig')->willReturn(['search' => ['engine' => 'legacy']]);
```
first is better for me